src/Components/FormComboBox.js

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { comboBox } from '../../node_modules/uswds/src/js/components';
+
+import { ValidationState } from '../Models/PatientFieldData';
+import ValidationMessage from './ValidationMessage';
+
+function FormComboBox({ label, property, required, onChange, options, showRequiredHint, size, validationState, value }) {
+  const ref = useRef();
+  const [focused, setFocused] = useState(false);
+  const [customOption, setCustomOption] = useState(null);
+
+  useEffect(() => {
+    const { current } = ref;
+    comboBox.on(current);
+    // manually add event handlers to the custom input added by USWDS
+    const input = current.querySelector('input[type="text"]');
+    input.addEventListener('input', (e) => setCustomOption(e.target.value));
+    input.addEventListener('focus', () => setFocused(true));
+    input.addEventListener('blur', () => setFocused(false));
+    input.value = value;
+    return () => {
+      comboBox.off(current);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const { current } = ref;
+    // manually add/remove validation classes to the custom input added by USWDS
+    const input = current.querySelector('input[type="text"]');
+    if (validationState === ValidationState.ERROR) {
+      input.classList.add('usa-input--error');
+    } else {
+      input.classList.remove('usa-input--error');
+    }
+    if (validationState === ValidationState.FIXED) {
+      input.classList.add('usa-input--success');
+    } else {
+      input.classList.remove('usa-input--success');
+    }
+  }, [validationState]);
+
+  // first check if value exists as customOption or in options, if not, set as the custom option
+  if (value && value !== customOption) {
+    if (options.every((o) => value.localeCompare(o.props.value, undefined, { sensitivity: 'base' }) !== 0)) {
+      setCustomOption(value);
+    } else if (customOption) {
+      setCustomOption(null);
+    }
+  }
+  // combine the custom entered value in the input box with the options as needed
+  let combinedOptions = options;
+  if (customOption && options.every((o) => customOption.localeCompare(o.props.value, undefined, { sensitivity: 'base' }) !== 0)) {
+    combinedOptions = [
+      <option key={customOption} value={customOption}>
+        {customOption}
+      </option>,
+    ].concat(options);
+  }
```

Comment:
This code from lines ~45-60 checks if the user has typed in something that doesn't exist in the options list, and creates a new combined options list with that customOption at the beginning.